### PR TITLE
 Use tahoe-sites.api - Patch 04b: Remove UserOrganizationMapping from openedx

### PR DESCRIPTION
**Pull request by shadinaif**
## Change description

Use tahoe-sites.api - Patch 04b: Remove UserOrganizationMapping from openedx

## TODO
 - [ ] Manual testing as noted by Omar: https://github.com/appsembler/edx-platform/pull/1194#pullrequestreview-1051225789

## Type of change
- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

**Review by OmarIthawi — COMMENTED** [review]
@shadinaif please test the changes manually as noted inline.

**Comment by OmarIthawi** [comment]
@shadinaif did you check the previous review? Please let me know if you've tested the steps on AMC and Dashboard locally?

File: openedx/core/djangoapps/appsembler/sites/permissions.py [hunk 1]
```diff
@@ -7,6 +9,10 @@ class AMCAdminPermission(permissions.BasePermission):
     """
 
     def has_permission(self, request, view):
-        is_organization_admin = request.user.userorganizationmapping_set.filter(is_amc_admin=True).exists()
-        is_superuser = request.user.is_superuser
-        return is_organization_admin or is_superuser
+        try:
+            is_organization_admin = is_active_admin_on_organization(
```
[the remark this answers is not in the document]
Comment (shadinaif) [reply openedx/core/djangoapps/appsembler/sites/permissions.py]: Thank you @OmarIthawi . I've removed the permission class `AMCAdminPermission` entirely and added a helper function to be used by `get_queryset`. Please review and let me know if I should revert this for any reason, or add tests to verify it ((already verified in devstack black-box testing))

File: openedx/core/djangoapps/user_authn/views/register.py [hunk 2]
```diff
@@ -536,22 +544,20 @@ def _handle_duplicate_email_username(self, request, data):
...
-                    # return ok. Since the user already exists, the next steps
-                    # will be to set the user as is_amc_admin=Ture and create
-                    # the OAuth tokens only.
-                    # TODO: In the AMC side, make sure to verify the user email
-                    #       before allowing to register it.
-                    mapping = mapping_query.get()
-                    mapping.is_amc_admin = True
-                    mapping.save()
-                    return self._create_response(request, {}, status_code=200)
+                try:
```
[the remark this answers is not in the document]
Comment (shadinaif) [reply openedx/core/djangoapps/user_authn/views/register.py:547]: same here

**Review by OmarIthawi — CHANGES_REQUESTED** [review]
@shadinaif thanks for your work here.

I suggest making a more gradual fix for the AMC permission issue.

This means this PR should not attempt to fix or improve the AMC permission in anyway, except for changing imports. Another PR should address that.

File: openedx/core/djangoapps/appsembler/sites/permissions.py [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+
 
 class AMCAdminPermission(permissions.BasePermission):
     """
     Allow making changes only if you're designated as an admin in AMC.
     """
 
     def has_permission(self, request, view):
-        is_organization_admin = request.user.userorganizationmapping_set.filter(is_amc_admin=True).exists()
+        is_organization_admin = UserOrganizationMapping.objects.filter(
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/sites/permissions.py:12]: The 1%, thanks!!

File: openedx/core/djangoapps/appsembler/sites/api.py [hunk 4]
```diff
@@ -233,9 +234,9 @@ def get(self, request):
         if user_email and organization_name:
             try:
                 organization = Organization.objects.get(name=organization_name)
-                mapping = UserOrganizationMapping.objects.get(user__email=user_email, organization=organization)
-                return Response({'username': mapping.user.username}, status=status.HTTP_200_OK)
-            except (Organization.DoesNotExist, UserOrganizationMapping.DoesNotExist):
+                user = get_organization_user_by_email(email=user_email, organization=organization)
+                return Response({'username': user.username}, status=status.HTTP_200_OK)
+            except User.DoesNotExist:
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/sites/api.py:239]: `ObjectDoesNotExist` catches all models and in this case we have two models: User and Organization.

```suggestion
            except ObjectDoesNotExist:
```

File: openedx/core/djangoapps/appsembler/sites/permissions.py [hunk 5]
```diff
@@ -1,12 +1,17 @@
 from rest_framework import permissions
 
+from tahoe_sites.models import UserOrganizationMapping
+
 
 class AMCAdminPermission(permissions.BasePermission):
     """
     Allow making changes only if you're designated as an admin in AMC.
     """
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/sites/permissions.py:9]:
```suggestion
    """
    
    # TODO: RED-2845 Remove this class when AMC is removed.
```

**Review by OmarIthawi — APPROVED** [review]
Thanks @shadinaif! Much better. I've added few minor changes. 

Please address them and this PR is good to go.

File: openedx/core/djangoapps/appsembler/sites/permissions.py (hunk 5 above)
Comment (shadinaif) [reply openedx/core/djangoapps/appsembler/sites/permissions.py:9]: Done

File: openedx/core/djangoapps/appsembler/sites/api.py (hunk 4 above)
Comment (shadinaif) [reply openedx/core/djangoapps/appsembler/sites/api.py]: Done

**Pull request merged**